app/models/calendar_event.rb

```diff
@@ -85,8 +85,10 @@ def self.project_date_filter(range, scope = Project)
   end
 
   def self.project_step_date_filter(range, scope = ProjectStep)
-    scope.where("actual_end_date BETWEEN :first AND :last OR scheduled_start_date BETWEEN :first AND :last "\
-      "OR old_start_date BETWEEN :first and :last", {first: range.first, last: range.last})
+    scope.
+    where("actual_end_date BETWEEN :first AND :last OR scheduled_start_date BETWEEN :first AND :last "\
+    "OR old_start_date BETWEEN :first and :last", {first: range.first, last: range.last}).
+    where(is_finalized: true)
```

Comment:
Let me check in with the client on this first to confirm that unfinalized steps are shown in loan calendar. Thank you!